static/js/stream_ui_updates.js

```diff
@@ -122,6 +122,16 @@ export function update_change_stream_privacy_settings(sub) {
     }
 }
 
+export function update_permissions_banner(sub) {
+    const $settings = $(`.subscription_settings[data-stream-id='${CSS.escape(sub.stream_id)}']`);
+    const banner_elem = $settings.find(".tip");
+    if (sub.can_change_stream_permissions) {
+        banner_elem.hide();
+    } else {
+        banner_elem.show();
+    }
```

Comment:
This feels hard to reason about with two variables in the template and one here, since `can_change_name_description` may no longer just depend on role in the future.  Maybe it'd be better to rework the new div as a 5-line template that gets rerendered when this function is called and the contents of the div into place?  

The other good option would be to just have both strings be present in the template and move the conditionals here entirely.